instagram/src/App.js

```diff
@@ -0,0 +1,153 @@
+import React, { Component } from 'react';
+import Fuse from 'fuse.js';
+import './App.scss';
+
+import dummyData from './dummy-data';
+import PostsPage from './components/PostContainer/PostsPage';
+import Authenticate from './components/Authentication/Authenticate';
+import LoginPage from './components/Authentication/LoginPage';
+
+const LoginAuthenticator = Authenticate(PostsPage)(LoginPage);
+
+class App extends Component {
+
+  constructor() {
+
+    super();
+
+    this.state = {
+
+      data: [],
+      displayedData: [],
+      username: '',
+      loggedIn: false
+
+    }
+
+  }
+
+  componentDidMount() {
+
+    let initialData = dummyData;
+
+    if (localStorage.data) {
+
+      initialData = JSON.parse(localStorage.data);
```

Comment:
It's probably preferred to use `getItem`. It doesn't make a difference here, but it could given a different situation (or even in this situation in the future). [Here](https://stackoverflow.com/a/15439059)'s a second opinion.